src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.DefaultProxyCredentials.cs

```diff
@@ -42,87 +43,90 @@ public void SetGet_Roundtrips()
             var wrongCreds = new NetworkCredential("wrongusername", "wrongpassword");
+            string expectCreds = "Basic " + Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes($"{rightCreds.UserName}:{rightCreds.Password}"));
 
-            using (HttpClientHandler handler = CreateHttpClientHandler())
-            using (var client = new HttpClient(handler))
+            await LoopbackServer.CreateServerAsync(async (proxyServer, proxyUrl) =>
             {
-                handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl, rightCreds);
-                handler.DefaultProxyCredentials = wrongCreds;
-
-                Task<HttpResponseMessage> responseTask = client.GetAsync(Configuration.Http.RemoteEchoServer);
-                Task<string> responseStringTask = responseTask.ContinueWith(t =>
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
                 {
-                    using (t.Result) return t.Result.Content.ReadAsStringAsync();
-                }, TaskScheduler.Default).Unwrap();
-                await (new Task[] { proxyTask, responseTask, responseStringTask }).WhenAllOrAnyFailed();
+                    handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl, rightCreds);
+                    handler.DefaultProxyCredentials = wrongCreds;
 
-                TestHelper.VerifyResponseBody(responseStringTask.Result, responseTask.Result.Content.Headers.ContentMD5, false, null);
-                Assert.Equal(Encoding.ASCII.GetString(proxyTask.Result.ResponseContent), responseStringTask.Result);
-
-                string expectedAuth = $"{rightCreds.UserName}:{rightCreds.Password}";
-                Assert.Equal(expectedAuth, proxyTask.Result.AuthenticationHeaderValue);
-            }
+                    Task<HttpResponseMessage> responseTask = client.GetAsync(Configuration.Http.RemoteEchoServer);
+                    await proxyServer.AcceptConnectionAsync(async connection =>
+                    {
+                        await connection.ReadRequestHeaderAsync().ConfigureAwait(false);
+                        if (connection.GetRequestHeaderValue("Proxy-Authorization") == null)
+                        {
+                            await connection.SendResponseAsync(HttpStatusCode.ProxyAuthenticationRequired, "Proxy-Authenticate: Basic\r\n");
+                            lines = await connection.ReadRequestHeaderAsync().ConfigureAwait(false);
+                        }
+                        // Verify that we got rightCreds instead of wrongCreds (or nothing).
+                        Assert.Equal(expectCreds, connection.GetRequestHeaderValue("Proxy-Authorization"));
+
+                        Task serverTask = connection.SendResponseAsync(HttpStatusCode.OK);
+
+                        await TestHelper.WhenAllCompletedOrAnyFailed(serverTask, responseTask).ConfigureAwait(false);
```

Comment:
Generally we don't use ConfigureAwait in test code.  I know we're using it in LoopbackServer in some places, but we should avoid it unless there's a good reason for it.